src/app/_components/Modal/Modal.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ClientPortal } from './ClientPortal';
+import { DefaultModalCloseButton } from './DefaultModalCloseButton';
+
+export const Modal = ({
+  isOpen,
+  onClose,
+  isBlurBackground = true,
+  isCloseButton = true,
+  children,
+}) => {
+  const blurBackground = (
+    <div className="fixed left-0 top-0 h-full w-full backdrop-blur-sm"></div>
+  );
+
+  return (
+    <ClientPortal selector="modal-root" show={isOpen}>
+      {isOpen && (
+        <>
+          {isBlurBackground && blurBackground}
+          <div
+            className="fixed left-0 top-0 flex h-full w-full items-center justify-center"
+            onClick={onClose}
+          >
+            {isCloseButton ? (
+              <DefaultModalCloseButton onClose={onClose}>
```

Comment:
🟠 looks a bit weird when you are passing modal content ( clihdren ) into close button component, is that intended?

Reply:
ok, changed to: ModalCloseButton